openapi-codegen/sbt-plugin/src/sbt-test/sbt-openapi-codegen/petstore/Expected.scala.txt

```diff
@@ -0,0 +1,388 @@
+    tags: Option[Seq[Tag]] = None,
+    id: Option[Long] = None,
+    photoUrls: Seq[String],
+    name: String,
+    category: Option[Category] = None
+  )
+
+  sealed trait PetStatus extends enumeratum.EnumEntry
+  object PetStatus extends enumeratum.Enum[PetStatus] with enumeratum.CirceEnum[PetStatus] {
+    val values = findValues
+    case object available extends PetStatus
+    case object pending extends PetStatus
+    case object sold extends PetStatus
+  }
+  case class Category (
+    id: Option[Long] = None,
+    name: Option[String] = None
+  )
+  case class Tag (
+    id: Option[Long] = None,
+    name: Option[String] = None
+  )
+  case class User (
+    userStatus: Option[Int] = None,
+    phone: Option[String] = None,
+    lastName: Option[String] = None,
+    id: Option[Long] = None,
+    username: Option[String] = None,
+    password: Option[String] = None,
+    email: Option[String] = None,
+    firstName: Option[String] = None
+  )
+  case class Order (
+    id: Option[Long] = None,
+    status: Option[OrderStatus] = None,
+    shipDate: Option[java.time.Instant] = None,
+    quantity: Option[Int] = None,
+    complete: Option[Boolean] = None,
+    petId: Option[Long] = None
+  )
+
+  sealed trait OrderStatus extends enumeratum.EnumEntry
+  object OrderStatus extends enumeratum.Enum[OrderStatus] with enumeratum.CirceEnum[OrderStatus] {
+    val values = findValues
+    case object placed extends OrderStatus
+    case object approved extends OrderStatus
+    case object delivered extends OrderStatus
+  }
+  case class ApiResponse (
+    code: Option[Int] = None,
+    `type`: Option[String] = None,
+    message: Option[String] = None
+  )
+  type FindPetsByTagsResponse <: Seq[Pet]
+  implicit val FindPetsByTagsResponseSeqDecoder: cats.xml.codec.Decoder[FindPetsByTagsResponse] = seqDecoder[Pet]("Pet", isWrapped = false).map(_.asInstanceOf[FindPetsByTagsResponse])
+  implicit val FindPetsByTagsResponseSeqEncoder: cats.xml.codec.Encoder[FindPetsByTagsResponse] =
+    seqEncoder[Pet]("Pet", isWrapped = false, itemName = "Pet").contramap(_.asInstanceOf[Seq[Pet]])
+  implicit val FindPetsByTagsResponseSeqSchema: sttp.tapir.Schema[FindPetsByTagsResponse] =
+    implicitly[Schema[Seq[Pet]]].map(x => Some(x.asInstanceOf[FindPetsByTagsResponse]))(_.asInstanceOf[Seq[Pet]])
+  type FindPetsByStatusResponse <: Seq[Pet]
+  implicit val FindPetsByStatusResponseSeqDecoder: cats.xml.codec.Decoder[FindPetsByStatusResponse] = seqDecoder[Pet]("Pet", isWrapped = false).map(_.asInstanceOf[FindPetsByStatusResponse])
+  implicit val FindPetsByStatusResponseSeqEncoder: cats.xml.codec.Encoder[FindPetsByStatusResponse] =
+    seqEncoder[Pet]("Pet", isWrapped = false, itemName = "Pet").contramap(_.asInstanceOf[Seq[Pet]])
+  implicit val FindPetsByStatusResponseSeqSchema: sttp.tapir.Schema[FindPetsByStatusResponse] =
+    implicitly[Schema[Seq[Pet]]].map(x => Some(x.asInstanceOf[FindPetsByStatusResponse]))(_.asInstanceOf[Seq[Pet]])
```

Comment:
These serdes probably look pretty whacky. The reason we need this for xml, rather than just a single schema as for json, is that the array schema can declare `xml: { wrapped: true }` which could lead to differing formats between endpoints. These type-scoped codecs (along with some subsequent mapping in the endpoint) permit for the individual-endpoint-scoped parameterisation that that requires